api/src/main/java/jakarta/persistence/EntityAgent.java

```diff
@@ -0,0 +1,266 @@
+     * @param id The id of the entity to retrieve
+     * @param lockMode The lock mode to apply to the entity
+     *
+     * @return a detached entity instance
+     */
+    <T> T get(Class<T> entityClass, Object id, LockModeType lockMode);
+
+    /**
+     * Retrieve a record, fetching associations specified by the
+     * given {@link EntityGraph}.
+     *
+     * @param graph The {@link EntityGraph}
+     * @param graphSemantic a {@link GraphSemantic} specifying
+     *                      how the graph should be interpreted
+     * @param id The id of the entity to retrieve
+     *
+     * @return a detached entity instance
+     *
+     * @since 6.3
+     */
+    <T> T get(EntityGraph<T> graph, Object id);
+
+    /**
+     * Retrieve a record, fetching associations specified by the
+     * given {@linkplain EntityGraph load graph}, and obtaining
+     * the specified {@link LockModeType lock mode}.
+     *
+     * @param graph The {@linkplain EntityGraph load graph}
+     * @param id The id of the entity to retrieve
+     * @param lockMode The lock mode to apply to the entity
+     *
+     * @return a detached entity instance
+     */
+    <T> T get(EntityGraph<T> graph, Object id, LockModeType lockMode);
+
+    /**
+     * Retrieve multiple rows, returning entity instances in a
+     * list where the position of an instance in the list matches
+     * the position of its identifier in the given array, and the
+     * list contains a null value if there is no persistent
+     * instance matching a given identifier.
+     *
+     * @param entityClass The class of the entity to retrieve
+     * @param ids         The ids of the entities to retrieve
+     * @return an ordered list of detached entity instances, with
+     *         null elements representing missing entities
+     */
+    <T> List<T> getMultiple(Class<T> entityClass, List<?> ids);
+
+    /**
+     * Refresh the entity instance state from the database.
+     *
+     * @param entity The entity to be refreshed.
+     */
+    void refresh(Object entity);
+
+    /**
+     * Refresh multiple entities.
+     *
+     * @param entities The entities to be refreshed.
+     *
+     * @see #refresh(Object)
+     */
+    void refreshMultiple(List<?> entities);
+
+    /**
+     * Refresh the entity instance state from the database.
+     *
+     * @param entity The entity to be refreshed.
+     * @param lockMode The LockMode to be applied.
+     */
+    void refresh(Object entity, LockModeType lockMode);
+
+    /**
+     * Fetch an association or collection that's configured for lazy loading.
+     * <pre>
+     * Book book = agent.get(Book.class, isbn);  // book is immediately detached
+     * agent.fetch(book.getAuthors());           // fetch the associated authors
+     * book.getAuthors().forEach(author -> ... );  // iterate the collection
+     * </pre>
+     *
+     * @param association a {@linkplain FetchType#LAZY lazy} association
+     */
+    void fetch(Object association);
```

Comment:
Usage might be more convenient if this returned the association, so you could do,
```
agent.fetch(book.getAuthors()).forEach(author -> ... );
```

```suggestion
     * @return the fetched association.
     */
    <T> T fetch(T association);
```